### PR TITLE
Use openstack project auth

### PR DESCRIPTION
**Pull request by vgramer**
### What this PR does / why we need it
In Openstack auth with tenant/TenantID are deprecated in favor of project/projectID. But these variables contain the same value.

This PR is the follow-up of kubermatic/kubermatic#8211 which is a prerequisite.

### Which issue(s) this PR fixes
fix kubermatic/kubermatic#6248

### Release note

```release-note
Add support for Openstack authentication with `project name` and  `project id`.  `tenant name`  and `tenant id`  are deprecated and have been renamed to `project name` and  `project id` in the UI
```

### CODE REVIEW

File: src/app/core/services/wizard/provider/openstack.spec.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  expect(req.request.headers.get(Openstack.Header.ProjectID)).toBeNull();
+  expect(req.request.headers.get(Openstack.Header.Username)).toEqual(FakeValue.Username);
+  expect(req.request.headers.get(Openstack.Header.Password)).toEqual(FakeValue.Password);
+  expect(req.request.headers.get(Openstack.Header.Domain)).toEqual(FakeValue.Domain);
+  expect(req.request.headers.get(Openstack.Header.ApplicationCredentialID)).toBeNull();
+  expect(req.request.headers.get(Openstack.Header.ApplicationCredentialSecret)).toBeNull();
+  expect(req.request.headers.get(Openstack.Header.Datacenter)).toEqual(FakeValue.Datacenter);
+}
+
+function expect_projectID_auth_header_to_be_defined(req: TestRequest) {
```
Comment (maciaszczykm) [inline src/app/core/services/wizard/provider/openstack.spec.ts:250]: Use camelCase for method names.

File: src/app/core/services/wizard/provider/openstack.spec.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+});
+
+enum FakeValue {
+  Username = 'the-user',
+  Password = 'fake-pass',
+  ApplicationCredentialID = 'the-application-id',
+  ApplicationCredentialSecret = 'the-ApplicationCredentialSecret',
+  Domain = 'the-domain',
+  Datacenter = 'the-datacenter',
+  Project = 'the-roject',
```
Comment (maciaszczykm) [inline src/app/core/services/wizard/provider/openstack.spec.ts:222]: It doesn't change anything but "p" is missing :)

File: src/app/core/services/wizard/provider/openstack.spec.ts (hunk 1 above)
Comment (vgramer) [reply src/app/core/services/wizard/provider/openstack.spec.ts]: done

File: src/app/core/services/wizard/provider/openstack.spec.ts (hunk 2 above)
Comment (vgramer) [reply src/app/core/services/wizard/provider/openstack.spec.ts]: nice catch :)

**Pull request merged**